### ISSUE TITLE
Enable Socks proxy support

### ISSUE THREAD

**Issue by rufusprice** [krille-chan/fluffychat#887]
### Feature Description

Enabling Android app to connect via socks proxy.

### Rationale

Letting user connect securely and privately without having to use device-wide VPN.

### Additional Context

Users who wish to connect to matrix network via tor usually need to download a tool like Orbot and set it as device-wide VPN. While this is enough in many cases, for users who don't want all the traffic going through the Tor network, there is no easy way of achieving this. Applications such as [Simplex chat](https://simplex.chat/docs/guide/app-settings.html) and [Amethyst Nostr client](https://github.com/vitorpamplona/amethyst/pull/357) enable the user to configure a socks proxy for all app traffic to be routed through it.

**Comment by erebion** [comment]
I agree, this is important. For some countries it is even necessary to use proxies.

**Comment by PRO-2684** [comment]
Even enabling device-wide VPN doesn't seem to work for me. Have to use Element Android because of this.

**Comment by kotauskas** [comment]
A stopgap solution is to use [the HTTP proxy support that's built into Dart's `HttpClient`](https://api.dart.dev/stable/3.11.0/dart-io/HttpClient/findProxy.html) by setting the `HTTPS_PROXY` environment variable. (It would also make sense to set `HTTP_PROXY` if not for the fact that a Matrix client certainly shouldn't be making any HTTP connections.) Note that this environment variable **requires an HTTP proxy** (i.e. HTTP `CONNECT`) and **a SOCKS5 proxy will not work!** If your proxy server is SOCKS5-only, you can use a tool like [`sthp`](https://github.com/KaranGauswami/socks-to-http-proxy) to get an HTTP proxy that uses a SOCKS5 proxy under the hood.

For example, if your SOCKS5 proxy server is on `[::1]:1080` and you want to allocate port 1081 to the HTTP `CONNECT` proxy, use `RUST_LOG=warn sthp --listen-ip ::1 --port 1081 --socks-address [::1]:1080` to start `sthp` and then run FluffyChat with the `HTTPS_PROXY` environment variable set to `http://[::1]:1081`.

**Comment by Squatch55** [comment]
Socks proxy support like how Telegram has would be a good feature. For example my friend in China cannot use any Matrix base app because of this. I hope this feature gets implemented.

**Comment by krille-chan** [comment]
Duplicate of #346 (proxy support)
